File: 122.best-time-to-buy-and-sell-stock-ii.java
/*
 * @lc app=leetcode id=122 lang=java
 *
 * [122] Best Time to Buy and Sell Stock II
 */

// @lc code=start
class Solution {
    public int maxProfit(int[] prices) {
        int answer = 0;

        for (int i = 0; i < prices.length - 1; i ++) {
            if (prices[i + 1] - prices[i] > 0) {
                answer += prices[i + 1] - prices[i];
            }
        }
        return answer;
    }
}
// @lc code=end

File: 122.best-time-to-buy-and-sell-stock-ii.py
#
# @lc app=leetcode id=122 lang=python3
#
# [122] Best Time to Buy and Sell Stock II
#

# @lc code=start
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        for index in range(len(prices) - 1):
            if prices[index + 1] - prices[index] > 0:
                profit += prices[index + 1] - prices[index]
        return profit
# @lc code=end

